service/storage_proxy.cc

```diff
@@ -483,17 +483,58 @@ class view_update_write_response_handler : public write_response_handler, public
                         std::move(targets), pending_endpoints, std::move(dead_endpoints), std::move(tr_state), stats, std::move(permit)) {
         register_in_intrusive_list(*p);
     }
+    ~view_update_write_response_handler();
 private:
     void register_in_intrusive_list(storage_proxy& p);
 };
 
 class storage_proxy::view_update_handlers_list : public bi::list<view_update_write_response_handler, bi::base_hook<view_update_write_response_handler>, bi::constant_time_size<false>> {
+    // _live_iterators holds all iterators that point into the bi:list in the base class of this object.
+    // If we remove a view_update_write_response_handler from the list, and an iterator happens to point
+    // into it, we advance the iterator so it doesn't point at a removed object. See #4912.
+    std::vector<iterator*> _live_iterators;
+public:
+    view_update_handlers_list() {
+        _live_iterators.reserve(10); // We only expect 1.
```

Comment:
why reserve more then?

Reply:
Fear, uncertainty, doubt? better to reserve 10 (the cost is miniscule) that be proven wrong in our expectations by a giant crash.

Reply:
ok, got it